### ISSUE TITLE
Feature request: Sieve tester

### ISSUE THREAD

**Issue by Daniel15** [mailcow/mailcow-dockerized#4030]
## Summary

See https://community.mailcow.email/d/745-sogo-filter-being-ignored

It is often difficult to debug long or complicated Sieve scripts, or determine exactly which rules would match. As an example, for my inbox I have over 25 rules, some of which overlap with each other (so multiple actions are performed on a single email). Most apps that can configure them (SOgo, Roundcube, etc) allow the rules to configured but do not provide a way to easily test the rules based on an email.

It would be nice to have a way to test a Sieve rule directly in Mailcow's UI. In this case, "test" means perform a dry run - Show what the rules would do, but don't actually perform any actions. Underneath the "[Show active user sieve filter]" link, there could be a "[Test sieve filter]" link. Clicking the link would provide a textarea where an email's source could be pasted. The interface would execute the Sieve rules and say what actions would be taken. I imagine it being similar to Fastmail's sievetest script: https://www.fastmail.com/cgi-bin/sievetest.pl

This could potentially just execute [`sieve-test`](https://manpages.debian.org/testing/dovecot-sieve/sieve-test.1.en.html) and pass through its output as-is.

## Motivation

Easier debugging of Sieve filters

## Additional context

N/A

**Comment by johanvandyck** [comment]
For those who are also struggling with sieve of waiting on this feature: a workaround
- export sieve filter. I use roundcube "export filter feature" for that in the cofiguartion section.
- export complete message of a mail (*.eml)
- user https://www.fastmail.com/cgi-bin/sievetest.pl to test your sieve filter with your mail.

another option is to use package sieve-testing in linux systems. See the https://www.systutorials.com/docs/linux/man/1-sieve-test/

It might be nice to facilitate this external testing in sogo if a complete dry run feature is not feasable.

Regards

Johan.
